be/src/pipeline/pipeline.h

```diff
@@ -98,13 +127,13 @@ class Pipeline : public std::enable_shared_from_this<Pipeline> {
 
 private:
     void _init_profile();
-    std::atomic<uint32_t> _complete_dependency;
 
     OperatorBuilders _operator_builders; // left is _source, right is _root
     OperatorBuilderPtr _sink;            // put block to sink
 
-    std::vector<std::weak_ptr<Pipeline>> _parents;
-    std::vector<std::shared_ptr<Pipeline>> _dependencies;
+    std::mutex _depend_mutex;
+    std::vector<std::pair<int, std::weak_ptr<Pipeline>>> _parents;
+    std::vector<std::pair<int, std::shared_ptr<Pipeline>>> _dependencies;
```

Comment:
why not a map ?

Reply:
the dependencies normally vary small，vector performance is better